airbyte-cdk/bulk/README.md

```diff
@@ -1,15 +1,107 @@
+  the BOM import.
+  Consequently, the declared version has a much higher chance of being the effective version
+  picked by gradle during dependency resolution.
+
+- The BOM import is re-exported by the `bulk-cdk-core-base` artifact meaning that the rest of the
+  CDK as well as connectors don't need to worry about Jackson version numbers either.
+
+It gets better when multiple BOMs are involve.
+Consider for example Micronaut and Jackson: Micronaut also depends on Jackson.
+This can (and will!) cause dependency version conflicts; these are much easier to resolve by
+reconciling just two BOM versions.
+
+While BOMs are undoubtedly useful, let's still try to keep external dependencies to a minimum
+outside of tests.
+Less dependencies, less problems.
+
+## Developing
+
+Perhaps the most striking difference with the legacy java CDK from a connector DX perspective is
+that there are no facilities equivalent to `useLocalCdk = true`.
+
+This is deliberate and the intention here is to force the testing of CDK functionality to remain
+in the CDK.
+Recall that this is too often not the case in the legacy java CDK because it's simply not possible
+to do so there.
+
+The Bulk CDK is different.
+Dependency injection makes it possible to mock concrete implementation behavior realistically
+enough that Bulk CDK tests have entire fake connectors defined inside of them.
+
+There's no reason now not to first make changes to the CDK and publish those, and only then make
+downstream changes to a connector.
+
+If there's truly a need to develop both simultaneously, then the way to go may be to:
+1. do experimental development in the connector, keeping the CDK- and the connector-specific code
+   separate;
+2. once the CDK-specific code is reasonably mature, hoist it into the Bulk CDK and test it there;
+3. finally, publish those changes and have the connector depend on the latest Bulk CDK version.
+
+## Publishing
+
+While the CDK is incubating, its published version numbers are 0.X where X is the _build number_.
+This build number is monotonically increasing and is based on the maximum version value found on
+the [maven repository that the jars are published to](https://airbyte.mycloudrepo.io/public/repositories/airbyte-public-jars/io/airbyte/bulk-cdk/).
+
+Artifact publication happens via a [github workflow](../../.github/workflows/publish-bulk-cdk.yml)
+which gets triggered by any push to the master branch, i.e. after merging a pull request.
+
+From a contributor's perspective, this means that there's no need to worry about versions or
+changelogs.
+From a client's perspective, just always use the latest version.
+
+Once the incubation period winds down and the CDK stabilizes, we can start thinking about contracts,
+semantic versioning, and so forth; but not until then.
+
+## Licensing
+
+The license for the Bulk CDK is Elastic License 2.0, as specified by the LICENSE file in the root
+of this git repository.
```

Comment:
cc @Hesperide:
I think this is the right license because:
1. We do want to allow community members to create fee/OSS connectors using the CDK
2. We con't want AWS stealing our code and competing with us

This is a departure from our [current CDK license, which is MIT ](https://github.com/airbytehq/airbyte/blob/master/airbyte-cdk/LICENSE)

Reply:
Thread started: https://airbytehq-team.slack.com/archives/C06E6CQMDT3/p1723132200231669